k-distribution/include/kframework/builtin/domains.md

```diff
@@ -1089,11 +1113,20 @@ module INT-KAST [kast]
 
 endmodule
 
-module INT-KORE [kore]
+module INT-KORE [kore, symbolic]
   imports private K-EQUAL
+  imports private BOOL
   imports INT-COMMON
 
-  rule I1:Int ==K I2:Int => I1 ==Int I2
+  rule I1:Int ==K I2:Int => I1 ==Int I2 [simplification]
+  rule {K1 ==Int K2 #Equals true} => {K1 #Equals K2} [simplification]
+  rule {true #Equals K1 ==Int K2} => {K1 #Equals K2} [simplification]
+  rule {K1 ==Int K2 #Equals false} => #Not({K1 #Equals K2}) [simplification]
+  rule {false #Equals K1 ==Int K2} => #Not({K1 #Equals K2}) [simplification]
+  rule {K1 =/=Int K2 #Equals true} => #Not({K1 #Equals K2}) [simplification]
+  rule {true #Equals K1 =/=Int K2} => #Not({K1 #Equals K2}) [simplification]
+  rule {K1 =/=Int K2 #Equals false} => {K1 #Equals K2} [simplification]
+  rule {false #Equals K1 =/=Int K2} => {K1 #Equals K2} [simplification]
```

Comment:
It seems to me we could get more mileage out of these rules by lifting all of them to `==K` instead, and then reversing the specialization of `==K => ==Int` to instead generalize `==Int => ==K`. Then we could have one set of equality reasoning rules that works for all of `==Int`, `==Bool`, and `==String`.

Reply:
The disadvantage of this is we would no longer be able to do SMT reasoning about equalities because we can't generate SMT for `==K` due to its dependence on sort injections to represent equalities.

Reply:
I mean, we can encode `I1 ==Int I2` as just `(= I1 I2)` in smt. If we use ==K instead, we cannot encode this in smt because we would have to encode something like `(= (kseq (inj I1) dotk) (kseq (inj I2) dotk))` and smt does not support parametric symbols like `inj`